alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -156,30 +158,66 @@ private function getOutputHash($o)
 		return $this->eTagHash;
 	}
 	
-	private function getAutoEmbedCode()
+	private function getAutoEmbedCode($targetId = null)
 	{
-		$config = json_encode($this->getRequestParameter("config"));
-		$entry_id = $this->getRequestParameter('entry_id');
-		$autoEmbedCode = "\n var player; var ovpProvider = new Providers.OvpProvider($this->partnerId,\"\",$config);\n" .
-			"\t    ovpProvider.getConfig(\"" . $entry_id . "\",$this->uiconfId).then(config => {\n" .
-			"\t    player = Playkit.playkit(config);\n" .
-			"\t }, \n" .
-			"\t err => {\n" .
-			"\t    console.log(err)\n" .
-			"\t})\n";
+		$targetId = $targetId ? $targetId : $this->getRequestParameter('targetId');
+		$hasTargetId = !is_null($targetId) && $targetId != "";
```

Comment:
Why add a parameter for the check simply check this "!is_null($targetId) && $targetId != """ inside the if